src/parsers/manifest/dash/parse_mpd.ts

```diff
@@ -275,16 +275,16 @@ function parseCompleteIntermediateRepresentation(
                   parsedPeriods[0]?.start !== undefined ? parsedPeriods[0].start :
                                                           0;
     let maximumTime : number | undefined;
-    if (contentEnd !== undefined) {
-      maximumTime = contentEnd;
-    } else if (mediaPresentationDuration !== undefined) {
+    if (mediaPresentationDuration !== undefined) {
       maximumTime = mediaPresentationDuration;
     } else if (parsedPeriods[parsedPeriods.length - 1] !== undefined) {
       const lastPeriod = parsedPeriods[parsedPeriods.length - 1];
       maximumTime = lastPeriod.end ??
                     (lastPeriod.duration !== undefined ?
                       lastPeriod.start + lastPeriod.duration :
                       undefined);
+    } else if (contentEnd !== undefined) {
+      maximumTime = contentEnd;
```

Comment:
Why? 

It makes sense to me to rely first on the maximum encountered position in the MPD (which if I understand correctly is what `contentEnd` is) for non-live contents.